backend/src/achievements/achievements.controller.ts

```diff
@@ -0,0 +1,56 @@
+import { AchievementsService } from './achievements.service';
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from '@nestjs/common';
+import { Public } from 'src/auth/auth.public.decorator';
+import { AchievementDto } from './dto/achievement.dto';
+import { ResponseAchievementDto } from './dto/response-achievement.dto';
+import { DeleteResult } from 'typeorm';
+
+@ApiTags('achievements')
+@Controller('achievements')
+export class AchievementsController {
+  constructor(private readonly achivementsService: AchievementsService) {}
+
+  @Public()
+  @Get()
+  async findAll(): Promise<ResponseAchievementDto[]> {
+    return await this.achivementsService.findAll();
+  }
+
+  @Public()
+  @Get(':id')
+  @ApiResponse({ status: 404, description: 'Record not found.' })
+  async findById(@Param('id') id: string): Promise<AchievementDto> {
+    const num: number = +id;
+
+    try {
+      return await this.achivementsService.findById(num);
+    } catch (EntityNotFoundError) {
+      throw new NotFoundException('Found no achievement with matching id.');
+    }
+  }
+
+  @Public()
```

Comment:
```suggestion
```
I think these routes should be protected to prevent unauthorized clients from modifying the achievements table . Probably the other routes can be protected as well, but creating and deleting achievements in particular